### PR TITLE
feat: rpc for getting all vaults registered by a specific accountId

### PR DESCRIPTION
**Pull request by sander2**

### CODE REVIEW

File: crates/vault-registry/rpc/src/lib.rs [hunk 1]
```diff
@@ -166,6 +173,20 @@ where
...
+        &self,
+        account_id: AccountId,
+        at: Option<<Block as BlockT>::Hash>,
+    ) -> JsonRpcResult<Vec<Vault>> {
+        let api = self.client.runtime_api();
+        let at = BlockId::hash(at.unwrap_or_else(|| self.client.info().best_hash));
+
+        handle_response(
+            api.get_vaults_by_account_id(&at, account_id),
+            "Enable to get vault ids".into(),
```
Comment (gregdhill) [inline crates/vault-registry/rpc/src/lib.rs:186]:
```suggestion
            "Unable to get vault ids".into(),
```

File: crates/vault-registry/src/types.rs [hunk 2]
```diff
@@ -147,7 +149,7 @@ impl Default for VaultStatus {
 }
 
 #[derive(Encode, Decode, Clone, PartialEq)]
-#[cfg_attr(feature = "std", derive(Debug))]
+#[cfg_attr(feature = "std", derive(Debug, serde::Serialize))]
```
Comment (gregdhill) [inline crates/vault-registry/src/types.rs:152]: Do we also need to derive `Deserialize`?

File: crates/vault-registry/rpc/src/lib.rs [hunk 3]
```diff
@@ -16,15 +16,20 @@ use sp_runtime::{
 use std::sync::Arc;
 
 #[rpc]
-pub trait VaultRegistryApi<BlockHash, VaultId, Balance, UnsignedFixedPoint, CurrencyId>
+pub trait VaultRegistryApi<BlockHash, VaultId, Balance, UnsignedFixedPoint, CurrencyId, AccountId, Vault>
```
Comment (gregdhill) [inline crates/vault-registry/rpc/src/lib.rs:19]: To be *super* pedantic:
```suggestion
pub trait VaultRegistryApi<AccountId, Balance, BlockHash, CurrencyId, VaultId, Vault, UnsignedFixedPoint>
```

Comment (sander2) [reply crates/vault-registry/rpc/src/lib.rs:19]: Haha wow, that _is_ super pedantic. We didn't sort them in the other apis, so I don't think it's worth spending the time to sort them here

File: crates/vault-registry/src/types.rs (hunk 2 above)
Comment (sander2) [reply crates/vault-registry/src/types.rs:152]: I'll check if the client requires it

Comment (sander2) [reply crates/vault-registry/src/types.rs]: It does

**Review by gregdhill — APPROVED** [review]

**Pull request merged**